src/edwards.rs

```diff
@@ -423,7 +427,41 @@ impl EdwardsPoint {
     /// Compress this point to `CompressedEdwardsY` format.
     pub fn compress(&self) -> CompressedEdwardsY {
         unimplemented!()
-    }    
+    }  
+
+    /// This function tries to build a Point over the Doppio Curve from
+    /// a `Y` coordinate and a Choice that determines the Sign o the `X`
+    /// coordinate that the user wants to use.
+    /// 
+    /// The function gets `X` by solving: 
+    /// `+-X = mod_sqrt((y^2 -1)/(dy^2 - a))`. 
+    /// 
+    /// The sign of `x` is choosen with a `Choice` parameter. 
+    /// 
+    /// For Choice(0) -> Negative result. 
+    /// For Choice(1) -> Positive result.
+    /// 
+    /// Then Z is always equal to `1`.
+    /// 
+    /// # Returns
+    /// `Some(EdwardsPoint)` if there exists a result for the `mod_sqrt`
+    /// and `None` if the resulting `x^2` isn't a QR modulo `FIELD_L`. 
+    pub fn new_from_y_coord(y: &FieldElement, sign: Choice) -> Option<EdwardsPoint> {
+        match ProjectivePoint::new_from_y_coord(&y, sign) {
+            None => return None,
+            Some(point) => return Some(EdwardsPoint::from(&point)),
+        }
+    } 
+
+    /// This function tries to build a Point over the Doppio Curve from
+    /// a random `Y` coordinate and a random Choice that determines the 
+    /// Sign o the `X` coordinate.
+    pub fn new_random_point() -> EdwardsPoint {
+        // Simply generate a random `ProjectivePoint`
+        // and once we get one that is valid, switch 
+        // it to Extended Coordinates.
```

Comment:
A link from here to the sage python code could be important if we want to specify how it can be done prior to the tests can help to make it more demonstrative.

Reply:
To which Python code?

Reply:
Since the Sage docs aren't pushed yet, you can pull this PR once merged and add the reference to your docs there. 

I think this will be the easiest since I don't know where you have referenced this.